src/mscorlib/src/System/String.Comparison.cs

```diff
@@ -1158,5 +1157,14 @@ public Boolean StartsWith(String value, Boolean ignoreCase, CultureInfo culture)
         }
 
         public bool StartsWith(char value) => Length != 0 && _firstChar == value;
+
+        public static void CheckStringComparison(StringComparison comparisonType)
+        {
+            // Single comparison to check if comparisonType is within [CurrentCulture .. OrdinalIgnoreCase]
+            if ((uint)(comparisonType - StringComparison.CurrentCulture) > (StringComparison.OrdinalIgnoreCase - StringComparison.CurrentCulture))
+            {
+                throw new ArgumentException(SR.NotSupported_StringComparison, nameof(comparisonType));
```

Comment:
Should this use ThrowHelper?

Reply:
We don't use ThrowHelper anywhere in this file. In what scenario would it help?

Reply:
ThrowHelper outside generics (where its codegen size); it helps with inlining and the size when inlined.

So 
```csharp
((uint)(comparisonType - StringComparison.CurrentCulture) > (StringComparison.OrdinalIgnoreCase - StringComparison.CurrentCulture))
``` 
will probably easily inline

but add `new`, `throw`, `nameof`, `SR.NotSupported_StringComparison` and it probably won't

Whereas `ThrowHelper.ThrowArgumentException(ExceptionResource.NotSupported_StringComparison)` is one `int` constant register load and a call with no post cleanup (as it only throws).

Reply:
Gotcha. So, it will help inline CheckStringComparison, mainly for when it won't end up throwing (which is the most common scenario).